src/main/java/uk/gov/justice/hmpps/prison/service/OffenderActivitiesService.java

```diff
@@ -1,38 +1,44 @@
 package uk.gov.justice.hmpps.prison.service;
 
 import org.springframework.data.domain.Page;
-import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
-import uk.gov.justice.hmpps.prison.api.model.OffenderActivities;
 import uk.gov.justice.hmpps.prison.api.model.OffenderActivitySummary;
+import uk.gov.justice.hmpps.prison.api.model.OffenderAttendance;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.Attendance;
 import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderProgramProfile;
 import uk.gov.justice.hmpps.prison.repository.jpa.model.ReferenceCode;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.AttendanceRepository;
 import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderProgramProfileRepository;
+import uk.gov.justice.hmpps.prison.security.VerifyOffenderAccess;
 import uk.gov.justice.hmpps.prison.service.support.LocationProcessor;
 
 import java.time.LocalDate;
 import java.util.List;
-import java.util.stream.Collectors;
 
 @Service
 @Transactional(readOnly = true)
 public class OffenderActivitiesService {
     private final OffenderProgramProfileRepository offenderProgramProfileRepository;
-    private final BookingService bookingService;
+    private final AttendanceRepository attendanceRepository;
 
-    public OffenderActivitiesService(final OffenderProgramProfileRepository offenderProgramProfileRepository, final BookingService bookingService) {
+    public OffenderActivitiesService(final OffenderProgramProfileRepository offenderProgramProfileRepository, final AttendanceRepository attendanceRepository) {
         this.offenderProgramProfileRepository = offenderProgramProfileRepository;
-        this.bookingService = bookingService;
+        this.attendanceRepository = attendanceRepository;
     }
 
-    public Page<OffenderActivitySummary> getStartedActivities(final String offenderNo, final LocalDate earliestEndDate, final PageRequest pageRequest) {
-        final var startedActivities = offenderProgramProfileRepository.findByNomisIdAndProgramStatusAndEndDateAfter(offenderNo, List.of("ALLOC", "END"), earliestEndDate,
-            pageRequest);
-        return startedActivities.map(this::transform);
+    @PreAuthorize("hasRole('SYSTEM_USER')")
```

Comment:
should this be on the resource to better document the role required at resource level

Reply:
There seems to be 2 conflicting conventions - annotation the services or the resources 🤷 . Ive added the role to the swagger description